### PR TITLE
Wr/esm

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?

ESM

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,7 @@ node_modules
 .idea
 
 oclif.manifest.json
+
+oclif.lock
```
Comment (mshanemc) [inline .gitignore:48]: only need 1

File: package.json [hunk 2]
```diff
@@ -34,37 +32,39 @@
...
+    "node": ">=18.0.0"
   },
   "files": [
     "bin",
     "/lib",
     "/messages",
-    "/oclif.manifest.json"
+    "/oclif.manifest.json",
+    "/oclif.lock",
+    "/oclif.lock"
```
Comment (mshanemc) [inline package.json:67]: only one

File: src/shared/npmCommand.ts [hunk 3]
```diff
@@ -105,11 +101,11 @@ class NpmCommand {
    */
   private static findNode(root?: string): string {
     const isExecutable = (filepath: string): boolean => {
-      if (osType() === 'Windows_NT') return filepath.endsWith('node.exe');
+      if (os.type() === 'Windows_NT') return filepath.endsWith('node.exe');
 
       try {
         if (filepath.endsWith('node')) {
-          // This checks if the filepath is executable on Mac or Linux, if it is not it errors.
+          // This  checks if the filepath is executable on Mac or Linux, if it is not it errors.
```
Comment (mshanemc) [inline src/shared/npmCommand.ts:108]: the next 4 changes looks like accidentally added spaces

File: test/shared/npmCommand.test.ts [hunk 4]
```diff
@@ -188,11 +199,13 @@ describe('should find the node executable', () => {
...
   it('finds node binary inside sfdx bin folder and runs npm show command', () => {
-    const npmMetadata = new NpmModule(MODULE_NAME, undefined, __dirname).show(DEFAULT_REGISTRY);
+    const npmMetadata = new NpmModule(MODULE_NAME, undefined, dirname(fileURLToPath(import.meta.url))).show(
+      DEFAULT_REGISTRY
+    );
     expect(accessSyncStub).to.have.been.calledOnce;
     expect(existsSyncStub).to.have.been.calledTwice;
     expect(osTypeStub).to.have.been.calledOnce;
-    expect(realpathSyncStub).to.have.been.calledOnce;
+    // expect(realpathSyncStub).to.have.been.calledOnce;
```
Comment (mshanemc) [inline test/shared/npmCommand.test.ts:208]: Were these a TODO, do the tests not work, has the logic changed, or do these need to be removed?

**Review by mshanemc — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (mdonnalley) [reply package.json:67]: related to oclif.lock - the `postpack` and `clean:lib` scripts should remove the `oclif.lock` too... Probably should add that to our migrate command

**Review by mshanemc — APPROVED** [review]

**Comment by mshanemc** [comment]
QA notes:

[made a commit with the dev-scripts bump to reduce devDeps and bumped sfdx-core]

./bin/dev.js plugins trust verify -n @salesforce/plugin-org
✅ shows that it is signed

./bin/dev.js plugins trust verify -n shane-sfdx-plugins
✅ shows not signed

[link into sf]
✅ installs
📓 note the double ESM warning
📓 also note that we're pointing the "open an issue about the warnings" at ourselves, too.  😄 

```
➜  plugin-trust git:(wr/esm) sf plugins install @salesforce/plugin-org
 ›   Warning: @salesforce/plugin-trust is a linked ESM module and cannot be auto-transpiled. Existing compiled source will be used instead.
 ›   Warning: @salesforce/plugin-trust is a linked ESM module and cannot be auto-transpiled. Existing compiled source will be used instead.
Polling for new version(s) to become available on npm... done
Successfully validated digital signature for @salesforce/plugin-org.
Finished digital signature check.
```

`sf plugins install shane-sfdx-plugins`
✅ prompts for verification
warnings and errors still return fine.

add plugin to allowList, reinstall
✅  `The plugin [shane-sfdx-plugins] is not digitally signed but it is allow-listed.`

**Pull request merged**